pkg/reconciler/taskrun/taskrun.go

```diff
@@ -207,28 +207,18 @@ func (c *Reconciler) updateStatusLabelsAndAnnotations(tr, original *v1alpha1.Tas
 	return nil
 }
 
-func (c *Reconciler) getTaskFunc(tr *v1alpha1.TaskRun) (resources.GetTask, v1alpha1.TaskKind) {
-	var gtFunc resources.GetTask
+// TODO: Use factory func instead of hard-coding this once OCI images are supported.
+func (c *Reconciler) getTaskResolver(tr *v1alpha1.TaskRun) (*resources.LocalTaskRefResolver, v1alpha1.TaskKind) {
+	resolver := &resources.LocalTaskRefResolver{
```

Comment:
Why not creating the `resolver` after the `if` ? the you could inline all.

```go
resolver := &resources.LocalTaskRefResolver{
	Namespace:    tr.Namespace,
	Tektonclient: c.PipelineClientSet,
	Kind: kind,
}
```

Reply:
True. This function is actually being deleted in the next PR. Both are very long and do different things so we wanted to split them up. We can still inline it if you want or we can wait until the next one.